localization/yabloc/yabloc_imgproc/src/graph_segment/graph_segment_core.cpp

```diff
@@ -0,0 +1,156 @@
+// Copyright 2023 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "yabloc_imgproc/graph_segment/graph_segment.hpp"
+#include "yabloc_imgproc/graph_segment/histogram.hpp"
+
+#include <opencv4/opencv2/highgui.hpp>
+#include <opencv4/opencv2/imgproc.hpp>
+#include <tier4_autoware_utils/system/stop_watch.hpp>
+#include <yabloc_common/cv_decompress.hpp>
+#include <yabloc_common/pub_sub.hpp>
+
+namespace yabloc::graph_segment
+{
+GraphSegment::GraphSegment()
+: Node("graph_segment"),
+  target_height_ratio_(declare_parameter<float>("target_height_ratio")),
+  target_candidate_box_width_(declare_parameter<int>("target_candidate_box_width"))
+{
+  using std::placeholders::_1;
+
+  // Subscriber
+  sub_image_ =
+    create_subscription<Image>("input/image_raw", 10, std::bind(&GraphSegment::on_image, this, _1));
+
+  pub_mask_image_ = create_publisher<Image>("output/mask_image", 10);
+  pub_debug_image_ = create_publisher<Image>("output/segmented_image", 10);
+
+  const double sigma = declare_parameter<double>("sigma");
+  const float k = declare_parameter<float>("k");
+  const int min_size = declare_parameter<double>("min_size");
+  segmentation_ = cv::ximgproc::segmentation::createGraphSegmentation(sigma, k, min_size);
+
+  // additional area pickup module
+  if (declare_parameter<bool>("pickup_additional_areas", true)) {
+    similar_area_searcher_ =
+      std::make_unique<SimilarAreaSearcher>(declare_parameter<float>("similarity_score_threshold"));
+  }
+}
+
+cv::Vec3b random_hsv(int index)
+{
+  double base = (double)(index)*0.7071;
```

Comment:
1/sqrt(2)?

Reply:
I totally forgot what that means. :sob: 
The colors used for visualization are not important, so I have simplified the formula in d9617f3.